Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -37,6 +39,20 @@ async function run() {
         if (!tl.exist(webDeployPkg)) {
             throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', webDeployPkg));
         }
+
+        var isParamFilePresentInPackage = false;
+        var isFolderBasedDeployment = !msDeployUtility.fileExists(webDeployPkg);
+        
+        if (!isFolderBasedDeployment) {
+            isParamFilePresentInPackage = await msDeployUtility.containsParamFile(webDeployPkg);
+        }
+
+        if(!tl.filePathSupplied('SetParametersFile')) {
+            setParametersFile = null;
```

Comment:
Can we move all the pkg validation logic to a function and return results to main using array like below
var results = function() {
..
..
return [a, b]
}
results[0]; // is a
results[1]; // is b